exporter/kafkaexporter/kafka_exporter.go

```diff
@@ -132,6 +135,29 @@ func newSaramaProducer(config Config) (sarama.SyncProducer, error) {
 	c.Producer.MaxMessageBytes = config.Producer.MaxMessageBytes
 	c.Producer.Flush.MaxMessages = config.Producer.FlushMaxMessages
 
+	for _, u := range []string{os.Getenv("HTTP_PROXY"), os.Getenv("HTTPS_PROXY")} {
```

Comment:
I would prefer not using variables fetched from the environment since it complicates testing it effectively and means there is two ways that the exporter can be configured which can complicate things since you can't provide a programatic transition if it changes in future.

This should be an updated field in the config that allows you to set a proxy setting as per my comment on the issue.

Reply:
I only really went with this as the otel docs state that any http(s) proxy ends set would be used. Kafka is probably unique enough to get an exception but it was a starting point. 

There wasn't any clear place to add a optional arg,  or a place higher up the call stack that would allow setting anything that could be checked/used in the sarama config  logic. Is there a common config that is accessible further down or how are optional config settings generally set?

Reply:
Looks like it can be done for you by enabling the fields defined here:

https://github.com/Shopify/sarama/blob/main/config.go#L113-L119

Reply:
That's what the PR is using? I thought you meant some config setting in this repo.

Reply:
So the sarama config will automatically look for the environment variables you're using or you can supply a custom proxy URL which your code is effectively doing. 

What I am suggesting is that we updated https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/kafkaexporter/config.go#L16-L40 


To either include:

```go
ProxyEnabled bool `mapstructure:"proxy_enabled"`
ProxyURL string `mapstructure:"proxy_url"`
```

From there, we just need to apply these from the collector config to the saram config.

Reply:
Ah ok. The `Config` struct looks perfect for the job.

Reply:
`ProxyEnabled` would be used for, consumes the default proxy environment variables , `ProxyURL` allows you to define an explicit proxy URL from config.

Granted you could just go:

```yaml
receivers:
  kafka:
    proxy_url: ${env:PROXY_URL}
```

I am okay if you don't include `ProxyEnabled`, it just means we should call out how to set the proxy value in the readme.

Reply:
No worries.  I'll update the README. 
Sarama didn't respect/use any HTTP(S)_PROXY envs set in the environment I was running the exporter in,  so I'm not exactly sure what envs they use?

Reply:
I'd agree that every component should use proxy env vars consistently, but you also need to implement handling of `NO_PROXY` in that case. Ideally, that should be done in the Sarama library itself. The collector codebase doesn't have any handling of proxy env vars, we just use `net/http` library. If we cannot do that, we should go with config options as @MovieStoreGuy suggested ignoring env vars.